app/services/data_migrations/backfill_opened_on_apply_at_from_audits.rb

```diff
@@ -0,0 +1,40 @@
+module DataMigrations
+  class BackfillOpenedOnApplyAtFromAudits
+    TIMESTAMP = 20210511084457
+    MANUAL_RUN = true
+
+    def change
+      Course
+        .where(open_on_apply: true)
+        .where('created_at < \'2020-04-01\'')
+        .update_all('opened_on_apply_at = \'2020-03-26\'')
```

Comment:
Why did we go with `created_at` less than `2020-04-01` instead of `2020-03-27`? 
I was also having a think, maybe going with when we added the `open_on_apply` field, to set the timestamp might be more accurate as you mentioned `2019-11-18` rather than when we started auditing?

Reply:
2020-03-26 is the timestamp from the commit and is approximate in that it does not tell us when the PR was merged and the code deployed. Another approach would be to set this timestamp for all open courses first, then overwrite it with information from audits, if available. The reason I haven't done this is I'd like to see if there are any records not covered by the current approach.

I think something around 2020-03-26 is more accurate than when the field was first introduced. This is because the default value for `open_on_apply` in the database was changed soon afterwards to `false` in another migration and because without the `audited` statement, introduced around 2020-03-26, we don't know when these courses were actually opened. 

I guess we could have a tiered approach, in which courses created before 2019-11-18 are set to 2019-11-18, courses before 2020-04-01 are set to 2020-03-26 and information from audits trumps all, if available. What do you think? Considering these courses are from a previous cycle, I don't think it matters that much what we choose.

Reply:
Thanks for the explanation, let's keep as is I reckon, best not overcomplicate it as like you said this data is old anyway.